File: python/2.BinaryDecimal/7.py
# revese integer

def reverse(x):
    INT_MAX =  pow(2, 31)
    INT_MIN =  pow(-2, 31)
    ans = 0
    while(x!=0):
        digit = x/10
        if(ans > INT_MAX/10): return 0
        if(ans < INT_MIN /10): return 0
        ans = ans*10+digit
        x /= 10

    return ans

res = reverse(121)
print(res)

File: C++/2.BinaryDecimal/7.cpp
#include <iostream>
using namespace std;

int reverse(int x) {
        int ans = 0;

        while(x!=0){
            int rem = x%10;
            if( ans > INT_MAX/10 ){
                return 0;
            }
            if( ans < INT_MIN/10 ){
                return 0;
            }
            ans = ans*10 + rem;
            x /= 10;
        }

        return ans;
}
int main() {
    
    cout << reverse(123) << endl;
    return 0;
}